### PR TITLE
fix: Added delay and logging to graceful shutdown (#6485)

### LINKED ISSUE #6485: Integration tests failed
* Link to run: https://github.com/keptn/keptn/actions/runs/1651940233
* Triggered by: Scheduled build
* Branch: master
* Version: 0.11.5-dev.202201031343
* Datetime: 202201031343
* Commit: 525d881a

Integration Tests have finished.

| Target Platform        |  Keptn Install    |  Keptn Auth  | Airgapped Images | Log Ingestion   | Log Forwarding   | Sequence States   | Sequence Timeout   | Sequence Control   | Sequence Queue   | Sequence Loop |  QG Standalone  |  Self Healing  |  Del-Assist  |  Backup-Restore  |  Resource Service  |  User-Man. Depl.  | Cont Deliv | Cont Deliv (Remote Exec) | Manage secrets | Webhook integration | Graceful Shutdown | Uniform Registration   |
| ---------------------- | ----------------- | ------------ | ---------------- | --------------- | ---------------- | ----------------- | ------------------ | ---------------- | ------------------ | ------------- | --------------- | -------------- | ------------ | ---------------- | ------------------ | ----------------- |----------- | ------------------------ | -------------- | ------------------- | ----------------- | ---------------------- |
| kubernetes/GKE 1.19 | success | success | skipped | success | success | success | success | success | success | success | success | success | success | success | success | success | skipped | success | success | success | success | success |
| kubernetes/GKE 1.21 | success | success | skipped | success | success | success | success | success | success | success | success | success | success | success | success | success | success | skipped | success | success | :x: **failure** | success |
| kubernetes/k3d-on-GHA v4.4.6 | success | success | success | skipped | skipped | skipped | skipped | skipped | skipped | skipped | skipped | skipped | skipped | skipped | success | skipped | skipped | skipped | skipped | skipped | skipped | skipped |
| kubernetes/k3s-on-GHA v1.16.15+k3s1 | success | success | skipped | success | success | success | success | success | success | success | success | success | skipped | skipped | success | skipped | skipped | skipped | success | success | skipped | success |
| kubernetes/k3s-on-GHA v1.21.1+k3s1 | success | success | skipped | success | success | success | success | success | success | success | success | success | skipped | skipped | success | skipped | skipped | skipped | success | success | skipped | success |
| openshift/minishift-on-GHA 3.11 | success | success | skipped | success | success | success | success | success | success | success | skipped | success | skipped | skipped | success | skipped | skipped | skipped | success | success | skipped | skipped |

Note: This issue was auto-generated from [integration_tests.yml](.github/workflows/integration_tests.yml)

### LINKED ISSUE #6501: Integration tests failed
* Link to run: https://github.com/keptn/keptn/actions/runs/1656613989
* Triggered by: Scheduled build
* Branch: master
* Version: 0.11.5-dev.202201041459
* Datetime: 202201041459
* Commit: 307abf91

Integration Tests have finished.

| Target Platform        |  Keptn Install    |  Keptn Auth  | Airgapped Images | Log Ingestion   | Log Forwarding   | Sequence States   | Sequence Timeout   | Sequence Control   | Sequence Queue   | Sequence Loop |  QG Standalone  |  Self Healing  |  Del-Assist  |  Backup-Restore  |  Resource Service  |  User-Man. Depl.  | Cont Deliv | Cont Deliv (Remote Exec) | Manage secrets | Webhook integration | Graceful Shutdown | Uniform Registration   |
| ---------------------- | ----------------- | ------------ | ---------------- | --------------- | ---------------- | ----------------- | ------------------ | ---------------- | ------------------ | ------------- | --------------- | -------------- | ------------ | ---------------- | ------------------ | ----------------- |----------- | ------------------------ | -------------- | ------------------- | ----------------- | ---------------------- |
| kubernetes/GKE 1.19 | success | success | skipped | success | success | success | success | success | success | success | success | success | success | success | success | success | skipped | success | success | success | success | success |
| kubernetes/GKE 1.21 | success | success | skipped | success | success | success | success | success | success | success | success | success | success | success | success | success | success | skipped | success | success | :x: **failure** | success |
| kubernetes/k3d-on-GHA v4.4.6 | success | success | success | skipped | skipped | skipped | skipped | skipped | skipped | skipped | skipped | skipped | skipped | skipped | success | skipped | skipped | skipped | skipped | skipped | skipped | skipped |
| kubernetes/k3s-on-GHA v1.16.15+k3s1 | success | success | skipped | success | success | success | success | success | success | success | success | success | skipped | skipped | success | skipped | skipped | skipped | success | success | skipped | success |
| kubernetes/k3s-on-GHA v1.21.1+k3s1 | success | success | skipped | success | success | success | success | success | success | success | success | success | skipped | skipped | success | skipped | skipped | skipped | success | success | skipped | success |
| openshift/minishift-on-GHA 3.11 | success | success | skipped | success | :x: **failure** | success | success | success | success | success | skipped | success | skipped | skipped | success | skipped | skipped | skipped | success | success | skipped | skipped |

Note: This issue was auto-generated from [integration_tests.yml](.github/workflows/integration_tests.yml)

### PR DESCRIPTION
**Pull request by RealAnna**
Signed-off-by: RealAnna <anna.reale@dynatrace.com>

## This PR

- adds logging and delay
- Integration test passed https://github.com/keptn/keptn/runs/4700660720?check_suite_focus=true#step:53:46 

Fixes #6485

### CODE REVIEW

**Review by thisthat — APPROVED** [review]
LGTM

File: test/go-tests/graceful_shutdown_test.go [hunk 1]
```diff
@@ -117,33 +116,39 @@ func Test_GracefulShutdown(t *testing.T) {
...
 	t.Log("Verify network access to public URI of helloservice in stage staging")
 	cartPubURL, err = GetPublicURLOfService(serviceName, keptnProjectName, "staging")
-	require.Nil(t, err)
+	logError(err, t, shipyardPod)
+
 	err = WaitForURL(cartPubURL+serviceHealthCheckEndpoint, time.Minute)
-	if err != nil {
-		t.Log(GetDiagnostics(shipyardPod, ""))
-	}
-	require.Nil(t, err)
```
Comment (mowies) [inline test/go-tests/graceful_shutdown_test.go:139]: No `require.Nil` at the end of the function even?

Comment (RealAnna) [reply test/go-tests/graceful_shutdown_test.go:139]: require nil is included in logError !

Comment (mowies) [reply test/go-tests/graceful_shutdown_test.go:139]: aha ok then :)

**Review by thisthat — APPROVED** [review]

**Pull request merged**